### ISSUE TITLE
Extend typescript-operations to support directives for operationResultType

### ISSUE THREAD

**Issue by larsenwork** [dotansimha/graphql-code-generator#6670]
**Is your feature request related to a problem? Please describe.**

In my frontend I'm consuming a schema that uses @deprecated directives — e.g. 

```GRAPHQL
type DemoThing {
  thingV1: string! @deprecated(reason: "thingV1 is old, use thingV2")
  thingV2: string!
}
```

which generates 

```typescript
export type DemoThing = {
  /** @deprecated thingV1 is old, use thingV2 */
  thingV1: string
  thingV2: string
}
```

and

```GRAPHQL
query getDemoThing {}


<!-- A clear and concise description of what the problem is. Ex. I'm always frustrated when [...] -->

**Describe the solution you'd like**
<!-- A clear and concise description of what you want to happen. -->

**Describe alternatives you've considered**
<!-- A clear and concise description of any alternative solutions or features you've considered. -->

**Additional context**
<!-- Add any other context or screenshots about the feature request here. -->

**Comment by ardatan** [comment]
I am not sure I understand your problem. Could you elaborate?

**Comment by larsenwork** [comment]
@ardatan thanks, and sure: 

### Current setup

Ultimately I'm fetching data with the hooks generated using `typescript-react-query`. In the example above I'd consume the hook like this 

```ts
const { data } = useGetDemoThingQuery()
```

the type for `data` will then be `GetDemoThingQuery | undefined` that doesn't have the JSDoc annotation inside it. So when in consume `data?.thingV1` it's not marked as deprecated. 

### My wish

That the `GetDemoThingQuery` includes the JSDoc comment so I can set up tooling to warn me when I consume `data?.thingV2`

**Comment by larsenwork** [comment]
So with 

```
export type GetDemoThingQuery = {
  /** @deprecated thingV1 is old, use thingV2 */
  thingV1: string
  thingV2: string
}
```

I can get 

![image](https://user-images.githubusercontent.com/3859425/133264046-a4b3529d-3dc0-4716-80f4-6cfa2383a100.png)

**Comment by Urigo** [comment]
@dotansimha @ardatan it seems reasonable to me, please let me know if it doesn't make sense?

@larsenwork want to start working on it?

**Comment by larsenwork** [comment]
@Urigo sure, it'd speed up my process greatly with some pointers where to start 👀 — i.e. if it's in the `typescript-operations` package or somewhere else.

**Comment by zomble** [comment]
I too have just realised deprecations are missing from typescript-operations, and that any deprecations wouldn't flow from api to consumers using this tooling.

**Comment by larsenwork** [comment]
@zomble yeah, it's a bit unfortunate. However without pointers I'd be a bit hard pressed to start fixing it but I could actually dedicate some office time to it this week if given a tiny pointer or two.
